### ISSUE TITLE
Replace usage of cPickle

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5136]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5136|
|Type|task|
|Created|2011-06-06 17:13:32Z|
|Branch|https://github.com/twisted/twisted/tree/remove-cpickle-5136|

From [Python 3k porting guide](http://docs.pythonsprints.com/python3_porting/py-porting.html#optimized-modules): Modules such as `pickle` and `StringIO` have traditionally been offered with C implementations for performance reasons. Rather than continue to expose two different implementations, a decision was made to expose both implementations under the same name, choosing to utilize the C implemenation if available and falling back to the pure-Python version if not.

This decision removes the need for the following idiom, instead making the more performant decision for the user.

```python
try:
    import cPickle as pickle
except ImportError:
    import pickle
```

The following files import `cPickle`:

```

./twisted/mail/relay.py:    import cPickle as pickle
./twisted/mail/relaymanager.py:    import cPickle as pickle
./twisted/persisted/dirdbm.py:    import cPickle as pickle
./twisted/persisted/journal/base.py:    import cPickle as pickle
./twisted/persisted/sob.py:    import cPickle as pickle
./twisted/plugin.py:        import cPickle
./twisted/plugin.py:        return cPickle
./twisted/test/test_persisted.py:    import cPickle as pickle
./twisted/test/test_twistd.py:import os, sys, cPickle, StringIO
./twisted/test/test_twistd.py:        cPickle.dump(service.Application("Hi!"), f)
```

Most of `twisted.persisted` has been, or will be, deprecated so those can be ignored.

All other imports of `pickle` can continue to exist (unless we want to speed them up under Python 2.x as well).

Attachments:

* [cPickle-5136.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/dfd/dfd71bbf509813d42882a245457908269b945c01/1fbadcad29290d019ba141e784f3d070274bb09c.patch) (791 bytes) - added by thijs on 2011-06-06 19:46:22Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5136 5136
type__task task
reporter__thijs thijs
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_remove_cpickle_5136 branches/remove-cpickle-5136
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1307380412000000 1307380412000000
changetime__1308503363000000 1308503363000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

The only import of cPickle which is not protected by a `try/except ImportError` is the one in `twisted/test/test_twistd.py`.  The rest should work fine on Python 3.x, since they catch `ImportError` and fall back to the `pickle` module.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_1):
> The only import of cPickle which is not protected by a `try/except ImportError` is the one in `twisted/test/test_twistd.py`.  The rest should work fine on Python 3.x, since they catch `ImportError` and fall back to the `pickle` module.

Thanks for the review. I attached a patch that adjusts the import.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess commented|
|-|-|

(In [32130]) Branching to 'remove-cpickle-5136'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for working on this, thijs. Looks good, please merge.

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-cpickle-5136)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [32145]) Merge remove-cpickle-5136: adjust cPickle import to make it compatible with Python 3.

Author: thijs
Reviewer: jesstess
Fixes: [#5136](https://github.com/twisted/twisted/issues/5136)
